File: backend/app/schemas/notification.py
import uuid
from typing import List, Optional
from pydantic import BaseModel
from datetime import datetime
from app.models.enums import NotificationType

class NotificationBase(BaseModel):
    title: str
    message: str
    notification_type: NotificationType
    
class NotificationCreate(NotificationBase):
    target_regions: Optional[List[str]] = None

class NotificationResponse(NotificationBase):
    id: uuid.UUID
    is_read: bool
    created_at: datetime
    priority: str = "Low"  # Inferred since not in DB
    health_risk: Optional[str] = None
    aqi: Optional[float] = None
    related_prediction_id: Optional[uuid.UUID] = None
    related_route_id: Optional[uuid.UUID] = None

class HealthAdvisoryResponse(BaseModel):
    id: uuid.UUID
    advisory_text: str
    risk_level: str
    created_at: datetime
    prediction_id: Optional[uuid.UUID] = None
    route_id: Optional[uuid.UUID] = None

class NotificationSummary(BaseModel):
    total_unread: int
    latest_notifications: List[NotificationResponse]

File: frontend/src/features/notifications/types/notification.ts
export type NotificationType = 
  | 'health_alert' 
  | 'route_recommendation' 
  | 'system_alert' 
  | 'daily_summary' 
  | 'weekly_summary' 
  | 'emergency';

export interface NotificationResponse {
  id: string;
  title: string;
  message: string;
  notification_type: NotificationType;
  is_read: boolean;
  created_at: string;
  priority: string;
  health_risk?: string;
  aqi?: number;
  related_prediction_id?: string;
  related_route_id?: string;
}

export interface HealthAdvisoryResponse {
  id: string;
  advisory_text: string;
  risk_level: string;
  created_at: string;
  prediction_id?: string;
  route_id?: string;
}

export interface UserPreferencesRead {
  preferred_language: string;
  preferred_aqi_unit: string;
  preferred_notification_method: string;
  preferred_dashboard_settings: string;
  preferred_theme: string;
  preferred_default_route_type: string;
  travel_preference: string;
  aqi_threshold?: number;
  notif_aqi_alerts: boolean;
}

export interface UserPreferencesUpdate {
  preferred_language?: string;
  preferred_aqi_unit?: string;
  preferred_notification_method?: string;
  preferred_dashboard_settings?: string;
  preferred_theme?: string;
  preferred_default_route_type?: string;
  travel_preference?: 'fastest' | 'safest' | 'balanced';
  aqi_threshold?: number;
  notif_aqi_alerts?: boolean;
}
